rllib/policy/eager_tf_policy.py

```diff
@@ -724,52 +661,130 @@ def set_state(self, state):
+                                    action_distribution_fn(
+                                        self, self.model,
+                                        input_dict[SampleBatch.OBS],
+                                        explore=explore,
+                                        timestep=timestep,
+                                        is_training=False)
+                            else:
+                                raise e
+                    elif isinstance(self.model, tf.keras.Model):
+                        input_dict = SampleBatch(input_dict, seq_lens=seq_lens)
+                        if state_batches and "state_in_0" not in input_dict:
+                            for i, s in enumerate(state_batches):
+                                input_dict[f"state_in_{i}"] = s
+                        self._lazy_tensor_dict(input_dict)
+                        dist_inputs, state_out, extra_fetches = \
+                            self.model(input_dict)
+                    else:
+                        dist_inputs, state_out = self.model(
+                            input_dict, state_batches, seq_lens)
 
-        @override(Policy)
-        def export_model(self, export_dir):
-            pass
+                    action_dist = self.dist_class(dist_inputs, self.model)
 
-        @override(Policy)
-        def export_checkpoint(self, export_dir):
-            pass
+                    # Get the exploration action from the forward results.
+                    actions, logp = self.exploration.get_exploration_action(
+                        action_distribution=action_dist,
+                        timestep=timestep,
+                        explore=explore)
+
+            # Action-logp and action-prob.
+            if logp is not None:
+                extra_fetches[SampleBatch.ACTION_PROB] = tf.exp(logp)
+                extra_fetches[SampleBatch.ACTION_LOGP] = logp
+            # Action-dist inputs.
+            if dist_inputs is not None:
+                extra_fetches[SampleBatch.ACTION_DIST_INPUTS] = dist_inputs
+            # Custom extra fetches.
+            if extra_action_out_fn:
+                extra_fetches.update(extra_action_out_fn(self))
+
+            return actions, state_out, extra_fetches
+
+        def _learn_on_batch_helper(self, samples):
+            # Increase the tracing counter to make sure we don't re-trace too
+            # often.
+            self.re_trace_counter += 1
+
+            with tf.variable_creator_scope(_disallow_var_creation):
+                grads_and_vars, _, stats = self._compute_gradients_helper(
+                    samples)
+            self._apply_gradients_helper(grads_and_vars)
+            return stats
 
         def _get_is_training_placeholder(self):
             return tf.convert_to_tensor(self._is_training)
 
         @with_lock
-        def _compute_gradients(self, samples):
+        def _compute_gradients_helper(self, samples):
```

Comment:
wait, why do we rename _compute_gradients to _compute_gradients_helper in the base class?
then you can't really @override(eager_policy_cls) these in TFEagerPolicy anymore?

also, should we increment re_trace_counter here as well?

Reply:
- Remember that these eager Policy classes are all automatically generated by RLlib. We don't really expect the user to write an eager Policy class, just a DynamicTFPolicy class via the `build_tf_policy()` utility, then this class get converted automatically into an eager policy via `DynamicTFPolicy.as_eager()`.
- Yes, we should increment here as well (same as in `_apply_gradients_helper`).